apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -1199,10 +1199,10 @@ namespace MWMechanics
         return false;
     }
 
-    void adjustDynamicStat(CreatureStats& creatureStats, int index, float magnitude)
+    void adjustDynamicStat(CreatureStats& creatureStats, int index, float magnitude, bool allowDecreaseBelowZero = false)
     {
         DynamicStat<float> stat = creatureStats.getDynamic(index);
-        stat.setCurrent(stat.getCurrent() + magnitude, index == 2);
+        stat.setCurrent(stat.getCurrent() + magnitude, allowDecreaseBelowZero);
```

Comment:
What about the case when `index == 2`? When this function is called  not for `DamageFatigue`? Now it always be not allowed to be negative. Not sure where it's expected.

You can make `allowDecreaseBelowZero  = true` by default. Change
```suggestion
        stat.setCurrent(stat.getCurrent() + magnitude, index == 2 && allowDecreaseBelowZero);
```
And call as:
```c++
adjustDynamicStat(creatureStats, effectKey.mId-ESM::MagicEffect::DamageHealth, -magnitude, uncappedDamageFatigue);
```

Reply:
adjustDynamicStat() is only called in effectTick(), and only for the following effects:
Restore effects (don't need to be uncapped)
Damage effects (already handled)
Absorb effects (they all need to be capped)
Sun damage (doesn't need to be uncapped, characters die when Health reaches 0)
Fire, Frost, Shock, Poison damage effects (don't need to be uncapped, characters die when Health reaches 0)

And we *are* only interested in uncapping for Damage Fatigue effect.

Limiting it for fatigue effects seems to be unnecessary, and I'm pretty sure assuming the "default" case is that no uncapping takes place is reasonable.